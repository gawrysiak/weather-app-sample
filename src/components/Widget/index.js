import * as React from 'react';
import { Card, Icon } from 'antd';

type Props = {
  data: {
    name: string,
    temperature: number,
    tempMin: number,
    tempMax: number,
    humidity: number,
    pressure: number,
  }
}

class Widget extends React.Component<Props> {
  render(): Card {
    return (
      <Card
        title={this.props.data.name}
        actions={[
          <Icon type="reload" />,
          <Icon type="delete" />,
        ]}
      >
        <p>
          Temp: {this.props.data.temperature}&#8451;<br />
          <small>{this.props.data.tempMin}&#8451; - {this.props.data.tempMax}&#8451;</small>
        </p>

        <p>Humidity: {this.props.data.humidity}</p>
        <p>Pressure: {this.props.data.pressure}</p>
      </Card>
    );
  }
}

export default Widget;
