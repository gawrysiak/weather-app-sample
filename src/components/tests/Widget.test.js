import React from 'react';
import renderer from 'react-test-renderer';
import Widget from 'components/Widget';

test('Widget component matches the snapshop', () => {
  const component = renderer.create(
    <Widget data={{
      temperature: 3,
      pressure: 1021,
      humidity: 99,
      tempMin: 1,
      tempMax: 4,
      name: 'Warsaw',
      icon: 'Snow',
    }} />,
  );

  expect(component.toJSON()).toMatchSnapshot();
});
