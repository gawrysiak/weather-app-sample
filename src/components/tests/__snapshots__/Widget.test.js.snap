// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget component matches the snapshop 1`] = `
<div
  className="ant-card ant-card-bordered"
>
  <div
    className="ant-card-head"
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        Warsaw
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={undefined}
  >
    <div>
      <p>
        Temp: 
        3
        ℃
        <br />
        <small>
          1
          ℃ - 
          4
          ℃
        </small>
      </p>
      <p>
        Humidity: 
        99
      </p>
      <p>
        Pressure: 
        1021
      </p>
    </div>
  </div>
  <ul
    className="ant-card-actions"
  >
    <li
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <span>
        <i
          className="anticon anticon-reload"
        />
      </span>
    </li>
    <li
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <span>
        <i
          className="anticon anticon-delete"
        />
      </span>
    </li>
  </ul>
</div>
`;
